.github/actions/setup-r-environment/action.yml

```diff
@@ -0,0 +1,55 @@
+name: 'setup R environment'
+description: 'Install R, Packages and necessary dependencies'
+author: 'Felix MIL'
+
+inputs:
+  install-pksim:
+    description: 'Wether to install PK-Sim'
+    required: true
+    default: 'false'
```

Comment:
also here: why are `instal-XXX` inputs defined as string and not as boolean?

Reply:
default value should be passed as string. It is evaluated as boolean in the workflow.

cf. https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputsinput_iddefault

Reply:
Not sure if the documentation is wrong or incomplete, because it works exactly as expected with passing the boolean values. E.g. check this example:
<details><summary>Reusable workflow <b>ReusableWithBooleanInput.yml</b></summary>
<p>

```
name: Reusable Workflow Example

on:
  workflow_call:
    inputs:
      my_boolean_input:
        description: 'A boolean input for the workflow'
        required: false
        type: boolean
        default: false

jobs:
  example_job:
    runs-on: ubuntu-latest
    steps:
      - name: Check input
        run: |
          echo "The boolean input is ${{ inputs.my_boolean_input }}"
          if ${{ inputs.my_boolean_input }}; then
            echo "The input is true"
          else
            echo "The input is false"
          fi

```
</p>
</details> 

<details><summary>Calling Workflow</summary>
<p>

```
name: Calling Workflow

on: push

jobs:
  call-true:
    uses: ./.github/workflows/ReusableWithBooleanInput.yml
    with:
      my_boolean_input: true
  call-false:
    uses: ./.github/workflows/ReusableWithBooleanInput.yml
    with:
      my_boolean_input: false
  call-default:
    uses: ./.github/workflows/ReusableWithBooleanInput.yml

```
</p>
</details> 

And the output is as expected:

<details><summary>Jobs output</summary>
<p>

<img width="240" src="https://github.com/Open-Systems-Pharmacology/Workflows/assets/25061876/678b3ec5-0a3b-4c07-b829-64d6817c0259"> <img width="240" src="https://github.com/Open-Systems-Pharmacology/Workflows/assets/25061876/df0f69bd-d58f-4765-9958-9dd214e43cef"> <img width="240" src="https://github.com/Open-Systems-Pharmacology/Workflows/assets/25061876/45f447c0-d28b-4bbf-8acf-b7afe6b0c00d">


</p>
</details> 

So I would suggest defining all boolean inputs explicitly with the type boolean and the default value as boolean and not as string

Reply:
This is a difference between Workflows and Actions.

Inputs were defined as booleans in workflows: https://github.com/Open-Systems-Pharmacology/Workflows/blob/9a0eb1c041679b7ab4dbee33a6832791ae336e6c/.github/actions/setup-r-environment/action.yml#L5-L9
But default values (set in actions) are defined as strings.

I tried to define them as boolean in reusable action but I had errors: https://github.com/Open-Systems-Pharmacology/Workflows/actions/runs/8064603776/job/22028820450#step:4:23

Reply:
I think it should work in the actions as well. Can it be that the error was due to omitted input type (which then was assumed as **string** as per default)?
So if you write e.g.
```
inputs:
  install-rClr:
    description: 'Wether to install rClr'
    required: true
    type: boolean
    default: false
```
in the action - do you still face the same error?

Reply:
It did not really make sense for me, why this should not work in an Action, so I tried it out 😄 - and in fact it works (s. the example below). 
[ExampleActionWithBooleanInput.zip](https://github.com/Open-Systems-Pharmacology/Workflows/files/14423239/ExampleActionWithBooleanInput.zip)

<details><summary>Action</summary>
<p>

```
name: 'Composite Action Example'
description: 'An example of a composite GitHub Action with a boolean input'

inputs:
  my_boolean_input:
    description: 'A boolean input for the action'
    required: false
    type: boolean
    default: false

runs:
  using: 'composite'
  steps:
    - name: Check input
      shell: bash
      run: |
        echo "The boolean input is ${{ inputs.my_boolean_input }}"
        if ${{ inputs.my_boolean_input }}; then
          echo "The input is true"
        else
          echo "The input is false"
        fi
```
</p>
</details> 

<details><summary>Using workflow</summary>
<p>

```
name: Test boolean Input via Action

on: push

jobs:
  action-call:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: ./.github/actions/test-boolean-input
        name: Call true
        with:
          my_boolean_input: true
      - uses: ./.github/actions/test-boolean-input
        name: Call false
        with:
          my_boolean_input: false
      - uses: ./.github/actions/test-boolean-input
        name: Call default
```
</p>
</details> 

<details><summary>Workflow output</summary>
<p>

![grafik](https://github.com/Open-Systems-Pharmacology/Workflows/assets/25061876/04b2db08-1110-4816-acb6-f3dc779a7319)


</p>
</details> 

However, the validator action fails and tells that the action is invalid

<details><summary>Validator action output</summary>
<p>

```
Treating action.yml as an Action definition
Validation failed: ValidationState {
    action_type: Some(
        Action,
    ),
    file_path: Some(
        "./.github/actions/test-boolean-input/action.yml",
    ),
    errors: [
        WrongType {
            code: "wrong_type",
            detail: Some(
                "The value must be string",
            ),
            path: "/inputs/my_boolean_input/default",
            title: "Type of the value is wrong",
        },
        Properties {
            code: "properties",
            detail: Some(
                "Additional property 'type' is not allowed",
            ),
            path: "/inputs/my_boolean_input",
            title: "Property conditions are not met",
        },
    ],
Fatal error validating ./.github/actions/test-boolean-input/action.yml
}
Error: Process completed with exit code 123.
```

</p>
</details> 

So let's keep it AS IS for now. As soon as GitHub fixes its documentation and action schema definition (which looks like a bug for me), we will be able to set the type properly.